### PR TITLE
Formatter returns a string or an object

### PR DESCRIPTION
**Pull request by edwardbr**
Hi slickgrid expects the formatter to be able to send an object or a string, if it is an object then you can specify a style as well as content.

### CODE REVIEW

**Comment by ghiscoding** [comment]
Hmm I've never heard of that, do you have some Slickgrid code to point to? 

Also, using `any` seems quite risky and will possibly open up a lot of future issues that could be totally unrelated (and I would rather not deal with them). If it's an object, doesn't it follow certain properties to use? If so, I would rather use them and change the type as something like `string | { style: string }`

File: src/app/modules/angular-slickgrid/models/formatter.interface.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 import { Column } from './column.interface';
 
-export type Formatter = (row: number, cell: number, value: any, columnDef: Column, dataContext: any, grid?: any) => string;
+export type Formatter = (row: number, cell: number, value: any, columnDef: Column, dataContext: any, grid?: any) => any;
```
Comment (ghiscoding) [inline src/app/modules/angular-slickgrid/models/formatter.interface.ts]: I cannot accept `any` Type, it's way too generic and will open a "can of worm".  You have provide me detail of what you said and also provide a sample code of what you are trying to achieve

**Comment by edwardbr** [comment]
I fully agree the benefits of TypeScript is type safety!  However the formatter element in a column definition can be either a string or an object as far as the core SlickGrid implementation is concerned.  
In this sample I am returning an object {addClasses: string, text: string}; so that I can specify different classes for each cell:

```
this.columnDefinitions.push({
            id: col.name,
            field: col.name,
            name: col.name,
            sortable: true,
            formatter: (row, cell, value, columnDef, rowData) => {
              const match = rowData.match;
              const state: ReconciliationItemState = match.state;

              let style = 'match';
              //other logic
              return {addClasses: style, text: value};
            }
          });
```

One thing that is really annoying me, (and this may be my lack of understanding in Angular) is that I would like different css styles for different instances of slickgrid on the same page.  I have a single panel containing a real time list of messages using a single column grid, and a central panel in this case containing a reconciliation report.  I am finding that each component's local scss is not working and I am limited to using the global scss instead.

**Comment by ghiscoding** [comment]
Can you update your PR now? I'm about to release another fix version and I will be done for quite some time

**Comment by edwardbr** [comment]
Hi I am a bit rusty on github UI, I have pushed my changes, I hope that it is reflected into this pull request, or shall I create a new one?

**Comment by ghiscoding** [comment]
You're missing the types on each properties. It should be this
```ts
string | { text: string; removeClasses: string; addClasses: string; }
```

**Comment by ghiscoding** [comment]
I went ahead and modified your PR directly in GitHub, I need to push my new version now so couldn't wait. Thanks for the PR and feedback

**Comment by ghiscoding** [comment]
Alright it's in the new version [2.1.3](https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v2.1.3), also note that after merging your PR, I had to do code refactoring in the ExportService because it uses the formatter and it wasn't just a string type anymore (ref [commit](https://github.com/ghiscoding/Angular-Slickgrid/commit/d8d4348a7937cd5e49bf95a97b073cef5d9b768e))

**Pull request merged**